packages/core/src/fhir-to-cda/cda-templates/components/medications.ts

```diff
@@ -0,0 +1,198 @@
+import {
+  buildCodeCe,
+  buildCodeCvFromCodeableConcept,
+  buildInstanceIdentifier,
+  formatDateToCdaTimestamp,
+  formatDateToHumanReadableFormat,
+  initiateSectionTable,
+  withoutNullFlavorObject,
+} from "../commons";
+import {
+  NOT_SPECIFIED,
+  _classCodeAttribute,
+  _codeAttribute,
+  _idAttribute,
+  _inlineTextAttribute,
+  _moodCodeAttribute,
+  _typeCodeAttribute,
+  _valueAttribute,
+  _xsiTypeAttribute,
+  extensionValue2014,
+  loincCodeSystem,
+  loincSystemName,
+  placeholderOrgOid,
+} from "../constants";
+import { createTableRowsAndEntries } from "../create-table-rows-and-entries";
+import { AugmentedMedicationStatement } from "./augmented-resources";
+
+const medicationsSectionName = "medications";
+const tableHeaders = ["Medication", "Dosage", "Frequency", "Start Date", "End Date", "Reason"];
+
+export function buildMedications(fhirBundle: Bundle) {
+  const medicationStatements: MedicationStatement[] =
+    fhirBundle.entry?.flatMap(entry =>
+      isMedicationStatement(entry.resource) ? [entry.resource] : []
+    ) || [];
+
+  if (medicationStatements.length === 0) {
+    return undefined;
+  }
+
+  const augmentedMedStatements = medicationStatements.map(statement => {
+    const ref = statement.medicationReference?.reference;
+    const refResource = ref ? findResourceInBundle(fhirBundle, ref) : undefined;
+    const medication = isMedication(refResource) ? refResource : undefined;
+    return new AugmentedMedicationStatement(statement, medicationsSectionName, medication);
+  });
+
+  const { trs, entries } = createTableRowsAndEntries(
+    augmentedMedStatements,
+    createTableRowsFromMedicationStatements,
+    createEntryFromStatement
+  );
+
+  const table = initiateSectionTable(medicationsSectionName, tableHeaders, trs);
+
+  const medicationsSection = {
+    component: {
+      section: {
+        templateId: buildInstanceIdentifier({
+          root: "2.16.840.1.113883.10.20.22.2.1.1",
+        }),
+        code: buildCodeCe({
+          code: "10160-0",
+          codeSystem: loincCodeSystem,
+          codeSystemName: loincSystemName,
+          displayName: "History of Medication use Narrative",
+        }),
+        title: "MEDICATIONS",
+        text: { table },
+        entry: entries,
+      },
+    },
+  };
+  return medicationsSection;
+}
+
+function isMedicationStatement(resource: Resource | undefined): resource is MedicationStatement {
+  return resource?.resourceType === "MedicationStatement";
+}
+
+function isMedication(resource: Resource | undefined): resource is Medication {
+  return resource?.resourceType === "Medication";
+}
+
+function createTableRowsFromMedicationStatements(
+  statement: AugmentedMedicationStatement,
+  medicationsPrefix: string
+): ObservationTableRow[] {
+  const trs: ObservationTableRow[] = [];
+  const tableRow = createTableRowFromObservation(statement, medicationsPrefix);
+  if (tableRow) trs.push(tableRow);
+  return trs;
```

Comment:
kinda odd to always create an array of one item or less. why not just return ObservationTableRow and modify the createTableRowsAndEntries to push the single object to the result instead of spreading the array?

Reply:
Fixed